contracts/contracts/CBOR.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.0;
+
+/// While parsing CBOR map, unexpected key
+error CBOR_Error_InvalidKey();
+
+/// While parsing CBOR map, length is invalid, or other parse error
+error CBOR_Error_InvalidMap();
+
+/// While parsing CBOR structure, data length was unexpected
+error CBOR_Error_InvalidLength(uint256);
+
+/// Value cannot be parsed as a uint
+error CBOR_Error_InvalidUintPrefix(uint8);
+
+/// Unsigned integer of unknown size
+error CBOR_Error_InvalidUintSize(uint8);
+
+/// CBOR parsed valid is out of expected range
+error CBOR_Error_ValueOutOfRange();
+
+error CBOR_Error_UintTooLong();
```

Comment:
Missing doc.

```suggestion
/// The size of Uint exceeds 8 bytes
error CBOR_Error_UintTooLong();
```

Reply:
This could also be considered a redundant comment, I haven't decided yet if it's worth implementing uint parsing for 128bit and 256bits.

Accepting this comment would make it more likely for the comments to become out of sync with functionality.

Lets decide, should we implement uint128 and uint256 parsing, or only handle 8,16,32 & 64 bits?

Reply:
Implementing uint128 and uint256 parsing would make sense, if there is a solidity<->solidity communication happening between the contracts in CBOR.

Reply:
The `cborg` library we use in Javascript doesn't support encoding/decoding of bigints beyond 64bit.

https://github.com/rvagg/cborg/blob/eca76d557991a4ded99bfe8ed062089b7bc5b21e/lib/0uint.js#L187

There is no Solidity <-> Solidity communication that needs uint128 or uint256 parsing.

Reply:
Maybe put this into the error comment here.

```solidity
/// @notice The size of Uint exceeds 8 bytes
/// @dev Since the CBOR libraries for Javascript don't support encoding/decoding
///      of bigints beyond 64bit, we decided not to implement parsing those.
error CBOR_Error_UintTooLong();
```

Reply:
I have added support for uint128 and uint256.

It's not using tagged bignums (https://www.rfc-editor.org/rfc/rfc8949.html#section-3.4.3), I need to double-check with https://github.com/oasisprotocol/oasis-cbor for compatibility.